rustc_tools_util/src/lib.rs

```diff
@@ -100,19 +104,42 @@ impl std::fmt::Debug for VersionInfo {
 }
 
 #[must_use]
-pub fn get_commit_hash() -> Option<String> {
-    let output = std::process::Command::new("git")
-        .args(["rev-parse", "HEAD"])
+pub fn rerun_if_git_changes() -> Option<()> {
+    // Make sure we get rerun when the git commit changes.
+    // First, find the `HEAD` file. This should work even with worktrees.
+    let output = Command::new("git").args(["rev-parse", "--git-dir"]).output().ok()?;
+    let stdout = output.status.success().then_some(output.stdout)?;
```

Comment:
We could use `git rev-parse --git-path HEAD` here to get the path and `git rev-parse --git-path [ref name]` below, that takes into account any environment variables that can change where things live

Reply:
Oh, I did not know about `--git-path`, neat!

Reply:
Hm, no that doesn't quite work. `--git-path` seems to just concatenate the given string after `.git`. So `git rev-parse --git-path [ref name]` prints `.git/[ref name]`, which is not where this ref is actually stored.

Reply:
Oh never mind, I misunderstood.

Reply:
All right, everything is using `--git-path` now.